modules/minecraft-discord-bot/ecs.tf

```diff
@@ -0,0 +1,60 @@
+resource "aws_ecs_service" "minecraft_discord_bot" {
+  name            = "${var.env}_minecraft_discord_bot"
+  cluster         = aws_ecs_cluster.minecraft_discord_bot.id
+  task_definition = aws_ecs_task_definition.minecraft_discord_bot.arn
+  launch_type = "FARGATE"
+  desired_count   = 1
+  network_configuration {
+    subnets = [var.ecs_subnet_id]
+    assign_public_ip = true
```

Comment:
Does this need a public IP?  I thought it just communicated externally with Discord via their API.

Reply:
I was having issues with running the bot without a public ip but I can try it again.

Reply:
Happy to take a look too.  I have made a discord bot or two in my day.

Reply:
So I'm pretty sure to not use a public ip I have to move the ECS service to a private subnet. So I went ahead and created the necessary resources for a private subnet and NAT gateway so the bot can continue to access the internet. I added the resources to the root `network.tf` file. Let me know if this is not what you were thinking.